app/services/unpublish_handler_service.rb

```diff
@@ -0,0 +1,89 @@
+class UnpublishHandlerService
+  UNPUBLISH_LOG_PATH = "#{Rails.root}/log/unpublish_message.log".freeze
+
+  def initialize(content_id)
+    @content_id = content_id
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    if email_params.empty?
+      nil
+    elsif taxon_is_unpublished?
+      email_ids = UnpublishEmailBuilder.call(email_params).ids
+      log_taxon_emails(email_ids)
+    else
+      log_none_taxon_lists(subscriber_lists)
+    end
+  end
+
+private
+
+  attr_reader :content_id
+
+  def taxon_is_unpublished?
+    subscriber_lists.first.links.keys.include?(:taxon_tree)
+  end
+
+  def queue_delivery_request_workers(email_ids)
+    email_ids.each do |email_id|
+      DeliveryRequestWorker.perform_async_in_queue(
+        email_id, queue: :delivery_immediate
+      )
+    end
+  end
+
+  def email_params
```

Comment:
I think sometimes it is clearer to explicitly pass in subscriber_lists so that you can see at the calling method that this depends on subscriber_lists. Same with some of the other methods

Reply:
I've done this for all the other methods, however i think for `email_params` it would look messy as i call `UnpublishEmailBuilder.call(email_params)` ? 

But noted that i should pass arguments more often.

Reply:
no I meant def ```email_params(subscriber_lists)``` so you know what stuff this method depends on

Reply:
Yep i know, i meant that because i call the method from `UnpublishEmailBuilder.call(email_params)`, it looks a little messy to me to have `UnpublishEmailBuilder.call(email_params(subscriber_lists))` but if you think it would make things clearer then i'm not opposed to changing it :)

Reply:
I've added the arguments.